### PR TITLE
Fix:Login failure causes Jackson serialization exception

### PR DESCRIPTION
**Pull request by gitdoit**
fix: #3695

### CODE REVIEW

**Comment by KomachiSion** [comment]
I think the beat solution is to find the reason of null key and solve it.

**Comment by gitdoit** [comment]
At the source of ` sun.net.www.MessageHeader` ,you can see

>An RFC 844 or MIME message header.  Includes methods
>for parsing headers from incoming streams, fetching
>values, setting values, and printing headers.
> **Key values of null are legal** : they indicate lines in
>the header that don't have a valid key, but do have
>a value (this isn't legal according to the standard,
>but lines like this are everywhere).

It can't be solved

File: client/src/main/java/com/alibaba/nacos/client/security/SecurityProxy.java [hunk 1]
```diff
@@ -143,7 +143,7 @@ public boolean login(String server) throws UnsupportedEncodingException {
                 HttpRestResult<String> restResult = nacosRestTemplate
                         .postForm(url, Header.EMPTY, Query.newInstance().initParams(params), bodyMap, String.class);
                 if (!restResult.ok()) {
-                    SECURITY_LOGGER.error("login failed: {}", JacksonUtils.toJson(restResult));
+                    SECURITY_LOGGER.error("login failed: {}", restResult.getMessage());
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/security/SecurityProxy.java:146]: Should there show username and password ?

Comment (gitdoit) [reply client/src/main/java/com/alibaba/nacos/client/security/SecurityProxy.java:146]: 登录失败后显示用户名密码有助于快速定位问题，但是会保存到日志里。或许，可以给一个更友好的提示。另外用户如果发现登录失败，首先肯定会去检查账号密码。

Comment (KomachiSion) [reply client/src/main/java/com/alibaba/nacos/client/security/SecurityProxy.java:146]: 这个日志里的用户名密码本身也就是用户配置的，保存在日志里问题应该不大。
至于更明显的提示，对应用来说也还是日志吧。总不能把程序杀掉吧。

Comment (KomachiSion) [reply client/src/main/java/com/alibaba/nacos/client/security/SecurityProxy.java:146]: 然后更友好的提示指的是 服务端的优化吗？ 比如用户不存在或者密码不正确？

Comment (gitdoit) [reply client/src/main/java/com/alibaba/nacos/client/security/SecurityProxy.java:146]: 首先，感谢你这么耐心的回复。你说的没错，日志文件也是面对用户的，账号密码也同样是。期待nacos越来越完善

**Pull request closed without merging**